etk-asm/src/asm.rs

```diff
@@ -430,203 +446,48 @@ impl Assembler {
                         }
                         .fail()
                     }
-                    Err(_) => {
-                        assert_eq!(self.pending_len, Some(0));
-                        self.pending_len = rop.size();
-                        self.pending.push_back(rop);
+                    Err(ops::Error::ContextIncomplete {
+                        source: UnknownLabel { label: _label, .. },
+                    }) => {
+                        let mut dynamic_push = false;
+                        match op.size() {
+                            Some(size) => self.concrete_len += size,
+                            None => {
+                                self.concrete_len += 2;
```

Comment:
If `op` is unsized, does it make sense to add two here? If I'm assembling something like:

```
%push(faraway)
%repeat_jumpdest(511)
faraway:
```

The push would have to be three bytes long.

Reply:
Yes, exactly, but in this instance it works more like a placeholder. Then, when the operations are actually concretised and the final position of the label is known, it is corrected. That's why also the dynamic jump flag is set to true. 

In fact, there is a test to prove that this works: 
https://github.com/gzanitti/etk/blob/12c609c081bc96812b067b12a45192c1f700e3c9/etk-asm/tests/asm.rs#L297